pkg/language/reader.go

```diff
@@ -428,6 +430,71 @@ func (r *Reader) ExitClassDecl(ctx *ClassDeclContext) {
 			typ.AddAttribute(memberCtx.GetResult())
 		}
 	}
+
+	if path := annotations.ReferencePath(typ); path != "" {
+		r.handleClassRef(typ, path)
+	}
+}
+
+func (r *Reader) handleClassRef(typ *concepts.Type, path string) {
+	if len(r.inputs) > 1 {
+		panic("refernced service with multiple inputs in undefined")
+	}
+
+	if r.service.Versions().Len() > 1 {
+		panic("cannot infer which version to add reference with multiple versions")
```

Comment:
Why not use the r.reporter.Errorf to report this errors?

Reply:
We can drop this anyway. Thanks!